packages/slice-machine/src/features/onboarding/helpers/useOnboardingProgress.ts

```diff
@@ -0,0 +1,70 @@
+import { Dispatch, SetStateAction, useEffect, useRef } from "react";
+
+import { onboardingSteps as steps } from "@/features/onboarding/content";
+import type {
+  OnboardingStep,
+  OnboardingStepStatuses,
+  OnboardingStepType,
+} from "@/features/onboarding/types";
+import { useLocalStorageItem } from "@/hooks/useLocalStorageItem";
+
+const getInitialState = (): OnboardingStepStatuses => {
+  // if the old guide was dismissed, all steps start as complete
+  const startComplete =
+    localStorage.getItem("slice-machine_isInAppGuideOpen") === "false";
+
+  return Object.fromEntries(
+    steps.map((step) => [step.id, startComplete]),
+  ) as OnboardingStepStatuses;
+};
+
+const useOnboardingStepStatus = (): [
+  OnboardingStepStatuses,
+  Dispatch<SetStateAction<OnboardingStepStatuses>>,
+] => {
+  const initialState = useRef(getInitialState()).current;
+  const [status, setStatus, { isUnset: isStatusUnset }] = useLocalStorageItem(
+    "onboardingSteps",
+    initialState,
```

Comment:
An initial state is not the same as a default value, so either the API for useLocalStorageItem is not the best for this or the usage is not accorded to the API.

Reply:
In this case, it was a default state (also the name of the hook prop), I was writing it as an initial state outside. I ended up changing the API and treating it as an initial state, and now the hook writes it to the storage if it's not defined already.

Reply:
Refactored to `usePersistedState`, this might still be relevant there